client/js/widgets/button.js

```diff
@@ -68,10 +68,28 @@ export class Button extends Widget {
       if(a.applyVariables) {
         if(Array.isArray(a.applyVariables)) {
           for(const v of a.applyVariables) {
-            if(v.parameter && v.variable)
+            if(v.parameter && v.variable) {
               a[v.parameter] = variables[v.variable];
-            else
+            }
+            else if (v.parameter && v.template) {
+              for (let key in variables) {
+                a[v.parameter] = v.template.replace(/({([^}]+)})/g, function(i) {
+                  let key = i.replace(/{/, '').replace(/}/, '');
+                  if (!variables[key]) return "";
```

Comment:
Any falsy value including zero in the variable will return "". Should probably actually check to see if the variable is defined instead.

Reply:
I second that.